lib/Itho/IthoCC1101.cpp

```diff
@@ -0,0 +1,852 @@
+  if (isRVHighCommand)   inIthoPacket.command = IthoHigh;
+  if (isMediumCommand)   inIthoPacket.command = IthoMedium;
+  if (isRVMediumCommand) inIthoPacket.command = IthoMedium;
+  if (isLowCommand)      inIthoPacket.command = IthoLow;
+  if (isRVLowCommand)    inIthoPacket.command = IthoLow;
+  if (isRVAutoCommand)   inIthoPacket.command = IthoStandby;
+  if (isStandByCommand)  inIthoPacket.command = IthoStandby;
+  if (isTimer1Command)   inIthoPacket.command = IthoTimer1;
+  if (isTimer2Command)   inIthoPacket.command = IthoTimer2;
+  if (isTimer3Command)   inIthoPacket.command = IthoTimer3;
+  if (isJoinCommand)     inIthoPacket.command = IthoJoin;
+  if (isJoin2Command)    inIthoPacket.command = IthoJoin;
+  if (isRVJoinCommand)   inIthoPacket.command = IthoJoin;
+  if (isLeaveCommand)    inIthoPacket.command = IthoLeave;
+
+#if defined (CRC_FILTER)
+  uint8_t mLen = 0;
+  if (isPowerCommand || isHighCommand || isMediumCommand || isLowCommand || isStandByCommand || isTimer1Command || isTimer2Command || isTimer3Command) {
+    mLen = 11;
+  }
+  else if (isJoinCommand || isJoin2Command) {
+    mLen = 20;
+  }
+  else if (isLeaveCommand) {
+    mLen = 14;
+  }
+  else {
+    return true;
+  }
+  if (getCounter2(&inIthoPacket, mLen) != inIthoPacket.dataDecoded[mLen]) {
+    inIthoPacket.command = IthoUnknown;
+    return false;
+  }
+#endif
+
+  return true;
+}
+
+bool IthoCC1101::checkIthoCommand(IthoPacket *itho, const uint8_t commandBytes[]) {
+  uint8_t offset = 0;
+  if (itho->deviceType == 28 || itho->deviceType == 24) offset = 2;
+  for (int i = 4; i < 6; i++)
+  {
+    //if (i == 2 || i == 3) continue; //skip byte3 and byte4, rft-rv and co2-auto remote device seem to sometimes have a different number there
+    if ( (itho->dataDecoded[i + 5 + offset] != commandBytes[i]) && (itho->dataDecodedChk[i + 5 + offset] != commandBytes[i]) ) {
+      return false;
+    }
+  }
+  return true;
+}
+
+void IthoCC1101::sendCommand(IthoCommand command)
+{
+  noInterrupts(); //Moved from plugin here to reduce blocking code
+  CC1101Packet outMessage;
+  uint8_t maxTries = sendTries;
+  uint8_t delaytime = 40;
+
+  //update itho packet data
+  outIthoPacket.command = command;
+  outIthoPacket.counter += 1;
+
+  //get message2 bytes
+  switch (command)
+  {
+    case IthoJoin:
+      createMessageJoin(&outIthoPacket, &outMessage);
+      break;
+
+    case IthoLeave:
+      createMessageLeave(&outIthoPacket, &outMessage);
+      //the leave command needs to be transmitted for 1 second according the manual
+      maxTries = 30;
+      delaytime = 4;
+      break;
+
+    default:
+      createMessageCommand(&outIthoPacket, &outMessage);
+      break;
+  }
+
+  //send messages
+  for (int i = 0; i < maxTries; i++)
+  {
+
+    //message2
+    initSendMessage(outMessage.length);
+    sendData(&outMessage);
+
+    finishTransfer();
+    delay(delaytime);
+  }
+  interrupts(); //Moved from plugin here to reduce blocking code
```

Comment:
The scope where interrupts are disabled is quite long here and one of the biggest NO-NO's during interrupts is called: `delay`
There is a number of messages created. Why do you need to disable interrupts for that?
And sending data while interrupts are disabled is also not what I consider a good design.

Please have a good look at this. Why need interrupts be disabled here?

Reply:
I've tested this at my side and it appears these sections of interrupts disabled code are not necessary at all and are remnants of the original version of the plugin made by Jodur. I will test some more to see if it results somehow in instability or missed commands, but likely to be removed in the next commit.

I will check with arjenhiemstra who made the current version of the Itho library if the things you mentioned above about the library can be resolved.